File: typescript/tests/components/inquiryForm.ts
import { Page } from "@playwright/test"

export class InquiryForm {
  constructor(private page: Page) {}

  async fillInquiryForm(inquiryData: {
    firstName: string
    lastName: string
    email: string
    phoneNumber: string
    message: string
  }) {
    await this.page.getByRole("textbox", { name: /first name/i }).fill(inquiryData.firstName)
    await this.page.getByRole("textbox", { name: /last name/i }).fill(inquiryData.lastName)
    await this.page.getByRole("textbox", { name: /email/i }).fill(inquiryData.email)
    await this.page.getByRole("textbox", { name: /phone number/i }).fill(inquiryData.phoneNumber)
    await this.page.getByRole("textbox", { name: /message/i }).fill(inquiryData.message)
    await this.page.getByRole("checkbox", { name: /i agree to receive marketing communications from DSwiss/i }).click()
    await this.page
      .getByRole("checkbox", {
        name: /I agree to allow DSwiss to store and process my personal data/i,
      })
      .click()
    await this.page.getByRole("button", { name: "submit" }).click()
  }
}

File: kotlin/src/test/kotlin/components/InquiryForm.kt
package components

import com.microsoft.playwright.Page
import com.microsoft.playwright.Page.GetByRoleOptions
import com.microsoft.playwright.options.AriaRole.*
import java.util.regex.Pattern
import java.util.regex.Pattern.CASE_INSENSITIVE

class InquiryForm(private val page: Page) {

    fun invoke(
        firstName: String,
        lastName: String,
        email: String,
        phoneNumber: String,
        message: String
    ) {
        page.getByRole(TEXTBOX, GetByRoleOptions().setName(Pattern.compile("first name", CASE_INSENSITIVE)))
            .fill(firstName)
        page.getByRole(TEXTBOX, GetByRoleOptions().setName(Pattern.compile("last name", CASE_INSENSITIVE)))
            .fill(lastName)
        page.getByRole(TEXTBOX, GetByRoleOptions().setName(Pattern.compile("email", CASE_INSENSITIVE)))
            .fill(email)
        page.getByRole(TEXTBOX, GetByRoleOptions().setName(Pattern.compile("phone number", CASE_INSENSITIVE)))
            .fill(phoneNumber)
        page.getByRole(TEXTBOX, GetByRoleOptions().setName(Pattern.compile("message", CASE_INSENSITIVE)))
            .fill(message)
        page.getByRole(
            CHECKBOX,
            GetByRoleOptions().setName(
                Pattern.compile("i agree to receive marketing communications from DSwiss", CASE_INSENSITIVE)
            )
        ).click()
        page.getByRole(
            CHECKBOX,
            GetByRoleOptions().setName(
                Pattern.compile("I agree to allow DSwiss to store and process my personal data", CASE_INSENSITIVE)
            )
        ).click()
        page.getByRole(BUTTON, GetByRoleOptions().setName("submit")).click()
    }
}